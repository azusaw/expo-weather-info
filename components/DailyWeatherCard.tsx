import React from "react";
import { StyleSheet } from "react-native";
import { DailyWeather } from "@/types";
import { Text, View } from "@/components/Themed";
import WeatherImage from "@/components/WeatherImage";
import dayjs from "dayjs";

const DailyWeatherCard = ({
  time,
  weatherCode,
  temperatureMin,
  temperatureMax,
}: DailyWeather) => (
  <View style={styles.card}>
    <Text style={styles.date}>{dayjs(time).format("DD ddd")}</Text>
    <WeatherImage weatherCode={weatherCode} />
    <Text style={styles.temperature}>
      {temperatureMin}
      <Text style={styles.slash}>/</Text>
      {temperatureMax}
      <Text style={styles.suffix}>â„ƒ</Text>
    </Text>
  </View>
);
export default DailyWeatherCard;

const styles = StyleSheet.create({
  card: {
    minWidth: 200,
    alignItems: "center",
    backgroundColor: "#fff",
    elevation: 3,
    marginLeft: 10,
    marginRight: 10,
    padding: 20,
    borderRadius: 10,
  },
  date: {
    textAlign: "center",
    fontSize: 24,
  },
  temperature: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    marginTop: 12,
    fontSize: 20,
    fontWeight: 100,
  },
  slash: {
    marginLeft: 10,
    marginRight: 10,
    fontSize: 24,
  },
  suffix: {
    marginLeft: 10,
    fontSize: 24,
  },
});
