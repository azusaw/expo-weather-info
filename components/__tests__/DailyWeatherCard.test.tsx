import { render } from "@testing-library/react-native";
import DailyWeatherCard from "@/components/DailyWeatherCard";

const mockProps = {
  time: "2025-04-16T23:45",
  weatherCode: 2,
  temperatureMin: 12.5,
  temperatureMax: 19.6,
};

describe("DailyWeatherCard", () => {
  test("it renders DailyWeatherCard successfully", () => {
    const { toJSON } = render(<DailyWeatherCard {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("it renders date in correct format", () => {
    const { getByText } = render(<DailyWeatherCard {...mockProps} />);
    expect(getByText("16 Wed")).toBeTruthy();
  });

  test("it renders correct weather description", () => {
    const { getByText } = render(<DailyWeatherCard {...mockProps} />);
    expect(getByText("Partly Cloudy")).toBeTruthy();
  });

  test(`it renders temperature`, () => {
    const { getByText } = render(<DailyWeatherCard {...mockProps} />);
    expect(
      getByText(`${mockProps.temperatureMin}/${mockProps.temperatureMax}`),
    ).toBeTruthy();
  });
});
