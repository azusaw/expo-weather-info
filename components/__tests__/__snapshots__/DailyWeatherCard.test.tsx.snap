// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeatherCard it renders DailyWeatherCard successfully 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "inherit",
      },
      {
        "alignItems": "center",
        "backgroundColor": "#ddd",
        "borderRadius": 10,
        "marginHorizontal": 10,
        "maxHeight": 180,
        "padding": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#888",
          "fontFamily": "Montserrat_400Regular",
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    16 Wed
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "inherit",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoImage
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={70}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={[]}
      source={
        [
          {
            "uri": "http://openweathermap.org/img/wn/02d@2x.png",
          },
        ]
      }
      style={
        {
          "height": 70,
          "width": 80,
        }
      }
      transition={
        {
          "duration": 1000,
        }
      }
      width={80}
    />
    <Text
      style={
        [
          {
            "color": "#888",
            "fontFamily": "Montserrat_300Light",
            "fontSize": 14,
          },
          undefined,
        ]
      }
    >
      Partly Cloudy
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#888",
          "fontFamily": "Montserrat_500Medium",
          "fontSize": 16,
        },
        {
          "marginTop": 10,
        },
      ]
    }
  >
    12.5
    <Text
      style={
        [
          {
            "color": "#888",
            "fontFamily": "Montserrat_200ExtraLight",
            "fontSize": 14,
          },
          {
            "marginHorizontal": 10,
          },
        ]
      }
    >
      /
    </Text>
    19.6
  </Text>
</View>
`;
